### PR TITLE
feat: add `context` option (`options.context`)

### LINKED ISSUE #148: [Feature] Provide default `context` option
Currently, `context` defaults to `compiler.options.context` from webpack. However, I'd like all my copy patterns to use `./src` as the default context, even though I can't change my webpack context to that for other reasons.

Could we add a fallback to the plugin options? The overall context fallback logic would be:

`context` property on pattern object -> `context` property in plugin options -> `context` property on webpack config

If this sounds okay, I can try to come up with a PR.

### PR DESCRIPTION
**Pull request by cletusw**
Overrides webpack context, overridden by pattern context

Fixes #148

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -46,6 +47,15 @@ function CopyWebpackPlugin(patterns = [], options = {}) {
         let contextDependencies;
         const written = {};
 
+        let context;
```
Comment (michael-ciniawsky) [inline src/index.js]: `\n` between the variable declaration and the if block please :)

Comment (cletusw) [reply src/index.js]: Should I place it up with the other `let`s and `const`s as well?

**Comment by michael-ciniawsky** [comment]
@cletusw Please rebase when time :)

**Comment by cletusw** [comment]
@michael-ciniawsky Awesome! Let me know when these patches get a version bump, thanks!

**Pull request merged**